src/scenario.jl

```diff
@@ -537,16 +535,11 @@ function run_model(domain::Domain, param_set::NamedDimsArray, corals::DataFrame,
         # Apply seeding
         if seed_corals && in_seed_years && has_seed_sites
             # Calculate proportion to seed based on current available space
-            seeded_area = (TA=n_TA_to_seed * col_area_seed_TA, CA=n_CA_to_seed * col_area_seed_CA)
             scaled_seed = distribute_seeded_corals(vec(total_site_area), prefseedsites, vec(leftover_space_m²), seeded_area)
 
-            # Seed each site with TA or CA
-            @views Y_pstep[seed_sc_TA, prefseedsites] .= Y_pstep[seed_sc_TA, prefseedsites] .+ scaled_seed.TA
-            @views Y_pstep[seed_sc_CA, prefseedsites] .= Y_pstep[seed_sc_CA, prefseedsites] .+ scaled_seed.CA
-
-            # Log seed values/sites (these values are relative to site area)
-            Yseed[tstep, 1, prefseedsites] .= scaled_seed.TA
-            Yseed[tstep, 2, prefseedsites] .= scaled_seed.CA
+            # Seed each site
+            @views Y_pstep[seed_sc, prefseedsites] .+= scaled_seed
+            Yseed[tstep, :, prefseedsites] .= scaled_seed
```

Comment:
This way, no need for that weird for loop.

FYI, when using loops in future, please conform to `idx` to indicate `index`.